resources/certificate.rb

```diff
@@ -18,11 +18,149 @@
 # limitations under the License.
 #
 
-actions :create, :delete, :acl_add
-default_action :create
-
-attribute :source, kind_of: String, name_attribute: true, required: true
-attribute :pfx_password, kind_of: String
-attribute :private_key_acl, kind_of: Array
-attribute :store_name, kind_of: String, default: 'MY', regex: /^(?:MY|CA|ROOT|TrustedPublisher|TRUSTEDPEOPLE)$/
-attribute :user_store, kind_of: [TrueClass, FalseClass], default: false
+include Windows::Helper
+
+property :source, String, name_property: true, required: true
+property :pfx_password, String
+property :private_key_acl, Array
+property :store_name, String, default: 'MY', regex: /^(?:MY|CA|ROOT|TrustedPublisher|TRUSTEDPEOPLE)$/
+property :user_store, [true, false], default: false
+
+action :create do
+  hash = '$cert.GetCertHashString()'
+  code_script = cert_script(true) <<
+                within_store_script { |store| store + '.Add($cert)' } <<
+                acl_script(hash)
+
+  guard_script = cert_script(false) <<
+                 cert_exists_script(hash)
+
+  converge_by("adding certificate #{new_resource.source} into #{new_resource.store_name} to #{cert_location}\\#{new_resource.store_name}") do
+    powershell_script new_resource.name do
+      guard_interpreter :powershell_script
+      convert_boolean_return true
+      code code_script
+      not_if guard_script
```

Comment:
This whole resource needs a going over, I've got an old note on some behavior with the .net classes used inside of it.  I'd rather get the existing behavior moved over and we can iterate from there.